src/Serilog.Sinks.Raygun/Sinks/Raygun/RaygunSink.cs

```diff
@@ -94,115 +106,122 @@ public void Emit(LogEvent logEvent)
             // Include the log level as a tag.
             var tags = _tags.Concat(new[] { logEvent.Level.ToString() }).ToList();
 
-            var properties = logEvent.Properties
-                         .Select(pv => new { Name = pv.Key, Value = RaygunPropertyFormatter.Simplify(pv.Value) })
-                         .ToDictionary(a => a.Name, b => b.Value);
+            var properties = logEvent.Properties.ToDictionary(kv => kv.Key, kv => kv.Value);
 
-            // Add the message
-            properties.Add("RenderedLogMessage", logEvent.RenderMessage(_formatProvider));
-            properties.Add("LogMessageTemplate", logEvent.MessageTemplate.Text);
+            // Add the message and template to the properties
+            properties[RenderedLogMessageProperty] = new ScalarValue(logEvent.RenderMessage(_formatProvider));
+            properties[LogMessageTemplateProperty] = new ScalarValue(logEvent.MessageTemplate.Text);
+            properties[OccurredProperty] = new ScalarValue(logEvent.Timestamp.UtcDateTime);
 
-            // Create new message
-            var raygunMessage = new RaygunMessage
+            // Add additional custom tags
+            if (properties.TryGetValue(_tagsProperty, out var eventTags) && eventTags is SequenceValue tagsSequence)
             {
-                OccurredOn = logEvent.Timestamp.UtcDateTime
-            };
+                tags.AddRange(tagsSequence.Elements.Select(t => t.ToString("l", null)));
 
-            // Add exception when available, else use the message template so events can be grouped
-            raygunMessage.Details.Error = logEvent.Exception != null
-                ? RaygunErrorMessageBuilder.Build(logEvent.Exception)
-                : new RaygunErrorMessage()
-                {
-                    ClassName = logEvent.MessageTemplate.Text,
-                    Message = logEvent.RenderMessage(_formatProvider),
-                    Data = logEvent.Properties.ToDictionary(k => k.Key, v => v.Value.ToString()),
```

Comment:
Note to reviewer: properties are already logged on the UserCustomData field, so removed the duplicates from the Error.Data field here.